packages/components/nodes/Notion/Notion.ts

```diff
@@ -0,0 +1,581 @@
+            {
+              label: 'Search all pages',
+              name: 'searchTarget',
+            },
+            {
+              label: 'Page',
+              name: 'pageTarget',
+            },
+            {
+              label: 'Database',
+              name: 'databaseTarget',
+            },
+            {
+              label: 'Block',
+              name: 'blockTarget',
+            },
+          ],
+        },
+        {
+          label: 'Page Operations',
+          name: 'pageOperation',
+          type: 'options',
+          options: [
+            {
+              label: 'Get Page',
+              name: 'getPage',
+            },
+            {
+              label: 'Create Page',
+              name: 'createPage',
+            },
+            {
+              label: 'Update Page',
+              name: 'updatePage',
+            },
+            {
+              label: 'Delete Page',
+              name: 'deletePage',
+            },
+          ],
+          default: 'Get Page',
+          show: {
+            'actions.target': ['pageTarget'],
+          },
+        },
+        {
+          label: 'Database Operations',
+          name: 'databaseOperation',
+          type: 'options',
+          options: [
+            {
+              label: 'Get Database',
+              name: 'getDatabase',
+            },
+            {
+              label: 'Create Database',
+              name: 'createDatabase',
+            },
+            {
+              label: 'Update Database',
+              name: 'updateDatabase',
+            },
+            {
+              label: 'Query Database',
+              name: 'queryDatabase',
+            },
+          ],
+          default: 'Get Database',
+          show: {
+            'actions.target': ['databaseTarget'],
+          },
+        },
+        {
+          label: 'Block Operations',
+          name: 'blockOperation',
+          type: 'options',
+          options: [
+            {
+              label: 'Get Block',
+              name: 'getBlock',
+            },
+            {
+              label: 'Update Block',
+              name: 'updateBlock',
+            },
+            {
+              label: 'Get Block Children',
+              name: 'getBlockChildren',
+            },
+            {
+              label: 'Append Block Children',
+              name: 'appendBlockChildren',
+            },
+            {
+              label: 'Delete Block',
+              name: 'deleteBlock',
+            },
+          ],
+          default: 'Get Block',
+          show: {
+            'actions.target': ['blockTarget'],
+          },
+        },
+      ] as INodeParams[];
+      this.inputParameters = [
+        {
+          label: 'Notion Version',
+          name: 'notionVersion',
+          type: 'string',
+          optional: false,
+          description: 'Notion Version to be used in the Headers of the request',
+        },
+        {
+          label: 'Page ID',
```

Comment:
In future we can turn these page ID/database ID into `asyncOptions`, and fetch the available IDs from `https://developers.notion.com/reference/post-search` and allow user to just select from dropdown instead of having them to put in the specific ID